storage/client/queries/queries.go

```diff
@@ -286,31 +286,82 @@ const (
 			ORDER BY id DESC`
 
 	ValidatorsData = `
+		WITH
+		-- Find all self-delegations for all accounts with active delegations.
+		self_delegations AS (
+			SELECT delegatee AS address, shares AS shares
+				FROM chain.delegations
+				WHERE delegator = delegatee
+				GROUP BY delegator, delegatee
+		),
+		-- Compute the number of delegators for each validator.
+		delegators_count AS (
+			SELECT delegatee AS address, COUNT(*) AS count
+				FROM chain.delegations
+				GROUP BY delegatee
+		),
+		-- Each entity can have multiple nodes. For each entity, find the validator node with the highest voting power.
+		validator_nodes AS (
+			SELECT entities.address, COALESCE(MAX(nodes.voting_power), 0) AS voting_power
+				FROM chain.entities AS entities
+				JOIN chain.nodes AS nodes ON
+					entities.id = nodes.entity_id AND
+					nodes.roles LIKE '%validator%'
+				GROUP BY entities.address
+		),
+		-- Compute validator rank by escrow balance active in a subquery to support querying a validator by address.
+		validator_rank AS (
```

Comment:
Should we sort/rank this list similarly as Oasis scan does by first listing all validators that are currently validating, and then those that don't (and within each of these two groups sorted by active escrow).

So something like:
```
order by is_validating, chain.accounts.escrow_balance_active DESC
```

Reply:
Good call, updated!

Reply:
Sorry, I forgot about this before, but maybe also active before non-active? 

```
ORDER BY is_validating DESC, active DESC, chain.accounts.escrow_balance_active DESC
```

That way entities with potentially a lot of stake that don't even intend to be validators (for whatever reason) will be after entities that are registered for validators but just don't have enough stake to be among top X.

Reply:
Ah good catch, agreed and updated!